bus-mapping/src/external_tracer.rs

```diff
@@ -1,9 +1,86 @@
 //! This module generates traces by connecting to an external tracer
-use crate::eth_types::{self, Address, GethExecStep, Word};
-use crate::BlockConstants;
+use crate::eth_types::{self, Address, Block, GethExecStep, Hash, Word, U64};
 use crate::Error;
 use geth_utils;
 use serde::Serialize;
+use std::str::FromStr;
+
+/// Definition of all of the constants related to an Ethereum block and
+/// chain to be used as setup for the external tracer.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
```

Comment:
Can't we derive `Copy`? Or does any of the internal field types not support it?

If we can, we could remove a lot of `.clone()`s for `BlockConstants`.

Reply:
`BlockConstants` is a struct with 8 fields.  I think it's better to make copying explicit with clone, because it's not small.  Also, this type is only used in tests.  And I think we only have one instance of clone for this type.

Reply:
Makes sense.